src/emulator/gdbstub/src/gdb.cpp

```diff
@@ -258,46 +293,74 @@ static std::string cmd_read_registers(HostState &state, PacketCommand &command)
     return stream.str();
 }
 
-static std::string cmd_write_registers(HostState &state, PacketCommand &command) { return ""; }
+static std::string cmd_write_registers(HostState &state, PacketCommand &command) {
+    if (state.gdb.current_thread == -1)
+        return "";
+    CPUState &cpu = *state.kernel.threads[state.gdb.current_thread]->cpu.get();
```

Comment:
This is not safe if the current_thread id doesn't point to a legitimate thread. Better make a null check

Reply:
I mean a check won't hurt, but gdb should be picking a valid thread anyway.

I will do it.

Reply:
I thought you can explicitly specify some random thread ID (in some case users doing extremely obscure thing). If it's too overload than drop, I'm really sorry.

Reply:
You could probably do that, and it's better to be safe than sorry. I can imagine someone accidentally typing an extra 1 at the end during an important debugging session and then crying their eyes out when everything suddenly crashes.

It's just that GDB should know all threads that exist, so it should stop them? I have no idea if that's the case or not though.